splink/linker.py

```diff
@@ -597,9 +599,14 @@ def _log_and_run_sql_execution(
         try:
             return self._run_sql_execution(final_sql, templated_name, physical_name)
         except Exception as e:
+            final_sql = sqlglot.parse_one(
+                final_sql,
+                read=self._sql_dialect,
+            ).sql(pretty=True)
+
```

Comment:
One thing I wondered about was what happens if `parse_one` throws an exception - for instance this (pretty contrived) example:

```py
import pandas as pd
from splink.duckdb.duckdb_linker import DuckDBLinker


df = pd.DataFrame([{"id": 1, "n": "a"}, {"id": 2, "n": "b"}])

linker = DuckDBLinker(df, {"link_type": "dedupe_only"})

linker.count_num_comparisons_from_blocking_rule("/* l.y = r.y )")
```

Not a huge deal but it means we don't end up with a `SplinkException`, which may have an impact if someone is writing code around this and is expecting to handle an exception of that type.

Reply:
But also this might point to something useful for helping with error messages - `sqlglot.errors.ParseError` helps point to _where_ in the SQL we are having an issue (in the case that it is a syntax thing), so may be something we could take advantage of in these sort of situations to help narrow things down, especially when the failing SQL is some massive block.
See also other possible blocking rules in above eg:
```py
"l.y ISS r.y"
"l.y === r.y"
"l.y === r.y"
```

Reply:
Yes, good point.

We could try plugging it through a `try` `except` clause and only pretty printing if it's parsable?

Reply:
yeah I think that sounds reasonable